DNN Platform/Website/Providers/DataProviders/SqlDataProvider/09.09.02.SqlDataProvider

```diff
@@ -0,0 +1,111 @@
+         WHERE IsNull(F.PortalID, -1) = IsNull(@PortalID, -1)
+           AND (F.FolderID   = @FolderID   OR IsNull(@FolderID,    -1) =  -1)
+           AND (F.FolderPath = @FolderPath OR IsNull(@FolderPath, N'') = N'')
+         ORDER BY F.PortalID, F.FolderPath
+         OPTION (OPTIMIZE FOR (@PortalID UNKNOWN));
+    ELSE BEGIN
+        --Retrieve Permission IDs
+            SELECT TOP 1 @AllPerm    = PermissionID FROM {databaseOwner}[{objectQualifier}Permission] WHERE PermissionCode = N'SYSTEM_FOLDER' AND ModuleDefId = -1 AND PermissionKey = N'WRITE';
+        IF IsNull(@Permissions, N'') LIKE N'%WRITE%'  SET @WritePerm = @AllPerm -- always checked, because WRITE always rules all other permissions
+        IF IsNull(@Permissions, N'') LIKE N'%READ%'
+            SELECT TOP 1 @ReadPerm   = PermissionID FROM {databaseOwner}[{objectQualifier}Permission] WHERE PermissionCode = N'SYSTEM_FOLDER' AND ModuleDefId = -1 AND PermissionKey = N'READ';
+        IF IsNull(@Permissions, N'') LIKE N'%BROWSE%'
+            SELECT TOP 1 @BrowsePerm = PermissionID FROM {databaseOwner}[{objectQualifier}Permission] WHERE PermissionCode = N'SYSTEM_FOLDER' AND ModuleDefId = -1 AND PermissionKey = N'BROWSE';
+        IF IsNull(@Permissions, N'') LIKE N'%ADD%' -- advanced permission provider only
+            SELECT TOP 1 @AddPerm    = PermissionID FROM {databaseOwner}[{objectQualifier}Permission] WHERE PermissionCode = N'SYSTEM_FOLDER' AND ModuleDefId = -1 AND PermissionKey = N'ADD';
+
+        SELECT * FROM {databaseOwner}[{objectQualifier}Folders]
+         WHERE (PortalID = IsNull(@PortalID, -1))
+           AND (FolderPath = @FolderPath OR IsNull(@FolderPath, N'') = N'')
+           AND (FolderID   = @FolderID   OR IsNull(@FolderID, -1) = -1)
+           AND FolderId IN -- granted folders
+                (SELECT DISTINCT P.FolderId FROM {databaseOwner}[{objectQualifier}FolderPermission] P
+                                       LEFT JOIN {databaseOwner}[{objectQualifier}UserPortalRoles](@PortalID, @UserID) R ON P.RoleID = R.RoleID
+                  WHERE (P.UserID = @UserID OR (R.RoleID Is Not Null))
+                    AND PermissionID IN (@AddPerm, @ReadPerm, @BrowsePerm, @AllPerm)
+                    AND AllowAccess = 1)
+           AND NOT FolderId IN -- denied folders
+                (SELECT DISTINCT P.FolderId FROM {databaseOwner}[{objectQualifier}FolderPermission] P
+                                       LEFT JOIN {databaseOwner}[{objectQualifier}UserPortalRoles](@PortalID, @UserID) R ON P.RoleID = R.RoleID
+                  WHERE (P.UserID = @UserID OR (R.RoleID Is Not Null))
+                    AND PermissionID IN (@AddPerm, @ReadPerm, @BrowsePerm, @WritePerm)
+                    AND AllowAccess = 0)
+         ORDER BY PortalID, FolderPath
+         OPTION (OPTIMIZE FOR (@FolderPath Unknown, @FolderId Unknown, @UserID Unknown));
+    END
+END -- Procedure
+GO
+
+DROP FUNCTION {databaseOwner}[{objectQualifier}UserPortalRoles]
```

Comment:
This needs to be wrapped in a conditional

Reply:
Thank you @mitchelsellers What is the minimum version of SQL the TSQL scripts for DNN need to be?  The reason I ask is, there is an [[IF EXISTS] clause to the DROP FUNCTION ](https://docs.microsoft.com/en-us/sql/t-sql/statements/drop-function-transact-sql?view=sql-server-ver15) with the statement "Available beginning with SQL Server 2016 and in SQL Database" in the Argument section.

Reply:
We [currently still support](https://www.dnndocs.com/content/getting-started/setup/requirements/index.html#database-server) SQL Server 2008 R2.

We may want to look at adjusting that at some point, though. [Entity Framework Core supports SQL Server 2012+](https://docs.microsoft.com/en-us/ef/core/providers/sql-server/#supported-database-engines). It looks like [Entity Framework 6 supports older SQL Server](https://stackoverflow.com/a/25240352/2688) (back to at least 2000).

Reply:
I would say for sure we keep it to our current minimums

Reply:
Extended Support for SQL Server 2008 R2 latet SU will end Juli 2022, and usually we drop support after that.
However, users are advised to use a product version with mainstream support - and given the fact that SQL Server Express Edition is sufficient for most small to medium DNN websites, there is in fact no reason not to upgrade or replace with a current release of SQL Server.